### ISSUE TITLE
The text for some options could be shortened

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#6345]
Feedback on GUI, received via email.

> Extra information in options text makes options very long and unwieldy
> 
> e.g.: The length of the text for the language option pushes the combo box far to the right of the dialog. This makes things seem miss-aligned to some extent and makes the dialog look messy. This is also true for the use saved settings button.  It looks out of proportion to the rest of the controls on the dialog. It also takes longer to move the mouse around these options. re-wording can assist, but some layout considerations could also help.
> 
> There are several ways to address this, see below: (option A is my prefered)
> 
> 1. remove the optional text in ( ).  Do we really need it given that users are prompted with dialogs or UAC anyway once they try to go ahead with the option? Note also that I think options are grayed out in the event that admin privilages are not available or if a portable copy is being used.  Perhaps we really do not need this extra text? If they change the language they get prompted to restart anyway.
> 
> 2. Place the optional text in the ( ) underneath the combo-box or button but ensure it is announced by NVDA properly 
> e.g.: 
> Language:  combo box  User default  collapsed 
> changes take effect after restarting NVDA
> 
> 3. In the case of the language option, leave the ( ) text as is but put the combo box underneath it instead of to the right (but closer to it's label text than the checkboxes under it).
> 
> 4. Remove the additional information in ( ) but include it as a tooltip on the combo box and button (ensuring NVDA still reads it). sighted users will see the tooltips.
> 
> 5. Use * and ** as a kind of legend and have the optional text at the bottom of the dialog above the OK/Cancel section 
> e.g.: 
> Language: **  combo box  User default  collapsed 
> Use NVDA on the Windows logon screen *  check box  not checked 
> Use currently saved settings on the logon and other secure screens *  button 
> and then at the bottom 
> \* requires administrator privileges 
> \*\* changes take effect after restarting NVDA 
> (You could do this in conjunction with option A so that this information is still visible in the dialog but where it does not get in the way of the flow of the options)

**Comment by feerrenrut** [comment]
I tend to agree with this. I think clarity is good, but if the text cant be shortend / removed, and long text is actually required then I would suggest we move most of it out of the label. Long text in buttons looks particularly strange, to the point that they may not be recognized as a button. I think using a legend should be reserved for cases where there are several options that require the legend. I think I have typically seen these kinds of explanatory text above the control they relate to rather than under, but perhaps I've seen it both ways.

**Comment by bhavyashah** [comment]
Nothing substantial in this comment, but just thought I'd invite comment from the training and support expert @Qchristensen on what the clearest, most concise, and aesthetic phrasing would be.

**Comment by Qchristensen** [comment]
Looking at NVDA's general options, the main settings which fit this are:

NVDA Language (Requires restart) - I tend to agree re removing (Requires restart) - yes it's true and most other options don't, but either you want to change language or not, and restarting NVDA generally wouldn't be that big an inconvenience - it's not like restarting the PC (which people may in fact confuse that with).  Actually0

The other two options on this screen with parentheses:
Use NVDA during sign-in (requires administrator privileges).
Use currently saved settings during sign-in and on secure screens (requires administrator privileges)

Perhaps we could put these two in their own "group" (the way that focus higlight and screen curtain have their own groups of options on the vision settings page), and the name of the group could be "Options requiring administrator privileges", we could then drop that text from the two individual options.

NVDA would read it as:
Options requiring administrator privileges groups, Use NVDA during sign-in, checked.

If we were going to do that, we might want to move that group of options to the bottom, unless you put the other options in their own groupings (as NVDA reports when you enter a grouping, but not when you leave one).


Back up in language, we list the name and the abbreviation - do we need the abbreviation? (English, en, French, fr, etc)  The only languages which have multiple options also have those options in plain text as well, eg:

Norwegian Bokmål (Norway) nb_NO
Norwegian Nynorsk (Norway) nn_NO
(and in the case of Norwegian, if both variants are primarily spoken in Norway... do we need to specify that?)

What makes that combo box so wide is actually Chinese:

Chinese (Simplified, China) zh_CN
Chinese (Traditional, Hong Kong, SAR) zh_HK
Chinese (Traditional, Taiwan) zh_TW

There's probably not a lot we can do to shorten those aside from remove the zh_CN zh_HK zh_TW etc, and possibly whether Hong Kong needs SAR at the end (I'm just looking at brevity, not politics, although isn't Taiwan a Special Autonomous Region as well?) - I assume that's down to how the translators have setup their individual languages?

Actually one other possible way of shortening a lot of these options:  This is the NVDA settings screen - do we need to specify "NVDA" for a lot of things in the options themselves?  EG the first four options are:
NVDA Language
Save configuration when exiting NVDA
show exit options when exiting NVDA
play sounds when starting or exiting NVDA

8 of 11 options on that page specify NVDA explicitly, even though (obviously) all options on the page are to do with NVDA.

**Comment by Adriani90** [comment]
I think it is clear enough that you have to restart NVDA after choosing a landuage and pressing enter since there is a dialog appearing saying that this is required. So the text in brackets can be removed.

**Comment by Adriani90** [comment]
Regarding the abbreviations of languages, they are fetched from Windows itself, I don't think we can do anything about it.
@CyrilleB79 do you have any thoughts on this?

**Comment by CyrilleB79** [comment]
@Adriani90  you wrote:
> I think it is clear enough that you have to restart NVDA after choosing a landuage and pressing enter since there is a dialog appearing saying that this is required. So the text in brackets can be removed.

I agree with this. I do not remember any software having such a label for UI language, even if most of thems also need a restart.

I have checked with MS Office that just call the option "Office display language", more or less because they allow many languages to be defined.
And when you validate the option dialog box, the following information message box appears:

>  Please restart Office so that your language changes can take effect.

with only an OK button.

Then you can continue working in Office with the old interface language. The new language will actually be used at next Office restart.

**Comment by CyrilleB79** [comment]
> Regarding the abbreviations of languages, they are fetched from Windows itself, I don't think we cannot remove them from the gui.
> @CyrilleB79 do you have any thoughts on this?

No, languages abbreviations are fetched from NVDA itself by listing the locale folder, i.e. all the languages in which NVDA is actually translated. Then full language names are provided by Windows from the language abbreviation. 
It is NVDA that makes the language list by concatenating full language name and language abbreviation. The choice to keep language abbreviation was made in order to be able to turn back to a known language after having set the interface to an unknown one. E.g. if you set 
interface language to Chinese and you cannot read that language, you will be able to restore interface language to English or German thanks to language abbreviation.
